### PR TITLE
RT#742040-Search page error

### PR DESCRIPTION
**Pull request by seenanair**
Key parameter was not provided to the Search page in Router which in turn caused the situation that xState machine is not fetching the right API

### CODE REVIEW

File: cypress/integration/pages/searchSpec.ts [hunk 1]
```diff
@@ -63,42 +63,15 @@ describe("Search", () => {
       }
     );
 
-    context("when a search has more results than the requested amount", () => {
```
Comment (khelwood) [inline cypress/integration/pages/searchSpec.ts:66]: Why is this test no longer required?

**Review by khelwood — COMMENTED** [review]
Seems to have fixed the problem. Can you explain the fix? My guess is some kind of stale data between the home and search pages.

**Comment by seenanair** [comment]
> Seems to have fixed the problem. Can you explain the fix? My guess is some kind of stale data between the home and search pages.

A key parameter is required for react-router for each the paths navigated. The Search page (as well as Home page) was missing that which was causing those issues. All the data which needs to pre-fetched before routing to a page was using a common state machine (xState) using a component DataFetcher. The default home page is fetching data using workTypes API and when we switch to Search page, since both of them don't have keys( which act as an identifier) the xState machine get confused and instead of calling the tissueTypes API , workTypes data is passed on Search page. The same was happening vice versa while visiting Home page from a Search page

File: cypress/integration/pages/searchSpec.ts (hunk 1 above)
Comment (seenanair) [reply cypress/integration/pages/searchSpec.ts:66]: Good check. This test is required.
 With the changes in Search page, the test cases using mock service data were failing  which took me a while to figure out and in the process of trying out different options, I think I somehow deleted it.  With Mock server 'once' API call, the mocked data was not returning which I didn't understand why. ('once' API  is used to mark the current request handler as used and to ignore any subsequent request matches to this handler. )

**Pull request merged**